docs/docs/visual-testing-with-storybook.md

```diff
@@ -21,97 +21,200 @@ npx -p @storybook/cli sb init
 
 This command adds a set of boilerplate files for Storybook in your project. However, since this is for a Gatsby project, you need to update the default Storybook configuration a bit so you don't get errors when trying to use Gatsby specific components inside of the stories.
 
-To update your Storybook config open `.storybook/config.js` and modify the content as follows:
+## Storybook version 5
 
-```js:title=.storybook/config.js
-import { configure } from "@storybook/react"
-import { action } from "@storybook/addon-actions"
+Storybook version 5.3 [brought a major change to how Storybook is configured](https://medium.com/storybookjs/declarative-storybook-configuration-49912f77b78).
+
+When you first install Storybook the only configuration file that will exist is `.storybook/main.js`, which will have the default stories location and default addons.
+
+```js:title=.storybook/main.js
+module.exports = {
+  stories: ["../stories/**/*.stories.js"],
+  addons: ["@storybook/addon-actions", "@storybook/addon-links"],
+}
+```
+
+Adjustments to Storybook's default `webpack` configuration are required so that you can transpile Gatsby source files and to ensure you have the necessary `babel` plugins to transpile Gatsby components. Add the following section to `.storybook/main.js`.
+
+```js:title=.storybook/main.js
+webpackFinal: async config => {
+    // Transpile Gatsby module because Gatsby includes un-transpiled ES6 code.
+    config.module.rules[0].exclude = [/node_modules\/(?!(gatsby)\/)/]
+
+    // use installed babel-loader which is v8.0-beta (which is meant to work with @babel/core@7)
+    config.module.rules[0].use[0].loader = require.resolve("babel-loader")
+
+    // use @babel/preset-react for JSX and env (instead of staged presets)
+    config.module.rules[0].use[0].options.presets = [
+      require.resolve("@babel/preset-react"),
+      require.resolve("@babel/preset-env"),
+    ]
+
+    config.module.rules[0].use[0].options.plugins = [
```

Comment:
I'm experiencing issues with the Storybook prop table not populating. 
Commenting out this line allows the PropTable to work but brings back GraphQL errors... any ideas?

Reply:
i've found that if i manually add `babel-plugin-react-docgen` the plugins the PropTable works again.. perhaps these options need to be spread before / after adding `@babel/plugin-proposal-class-properties` and `babel-plugin-remove-graphql-queries`


This is my final config:


```javascript
  webpackFinal: async config => {
    // Transpile Gatsby module because Gatsby includes un-transpiled ES6 code.
    config.module.rules[0].exclude = [/node_modules\/(?!(gatsby)\/)/];

    // use installed babel-loader which is v8.0-beta (which is meant to work with @babel/core@7)
    config.module.rules[0].use[0].loader = require.resolve("babel-loader");

    // use @babel/preset-react for JSX and env (instead of staged presets)
    config.module.rules[0].use[0].options.presets = [
      require.resolve("@babel/preset-react"),
      require.resolve("@babel/preset-env"),
    ];

    config.module.rules[0].use[0].options.plugins = [
      // use @babel/plugin-proposal-class-properties for class arrow functions
      require.resolve("@babel/plugin-proposal-class-properties"),

      // use babel-plugin-remove-graphql-queries to remove static queries from components when rendering in storybook
      require.resolve("babel-plugin-remove-graphql-queries"),

      // use babel-plugin-react-docgen to ensure PropTables still appear
      require.resolve("babel-plugin-react-docgen"),
    ];

    // Prefer Gatsby ES6 entrypoint (module) over commonjs (main) entrypoint
    config.resolve.mainFields = ["browser", "module", "main"];

    return config;
  },

```

Reply:
@PaulieScanlon have you tried issuing a build, a development build for instance before running Storybook? More on that in #20624.

Reply:
Yeah i did run in to issues if i'd recently cleaned Gatsby so running a dev first and then Storybook with the doc-gen back in config seems to work but i have noticed it blats out all other webpack config so is there a way to spread the plugins before applying the babel plugins so it maintains some of the original config?

Reply:
@PaulieScanlon If that issue is still persisting with these changes applied, can you open a new issue to track it? Thanks!